app/src/main/java/com/duckduckgo/app/global/DuckDuckGoApplication.kt

```diff
@@ -205,6 +213,18 @@ open class DuckDuckGoApplication : HasAndroidInjector, Application(), LifecycleO
         }
     }
 
+    private suspend fun moveUserToEstablished3DaysAfterInstall() {
```

Comment:
Instead of adding this logic to the `Application` class, you could add this logic to `UserStageStore` or another class. Then implement `LifecycleObserver` and add the class to the observers (line `185`).
With that we can remove the logic to migrate users on day 3 from the Application class, I think it will be cleaner.

Reply:
I've moved it as suggested to the UserStageStore 👍